tests/test_BestSeeingWcsSelectImages.py

```diff
@@ -0,0 +1,161 @@
+        self.localSetUp(mockFWHMs=[])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), 0)
+
+    def testSelectOne(self):
+        self.config.nImagesMax = 1
+        self.localSetUp(mockFWHMs=[self.avgFwhm for i in range(self.config.nImagesMax)])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), 1)
+        self.assertEqual(result.exposureInfoList[0].dataId, self.dataId)
+
+    def testSelectDefault(self):
+        """Test that the default configuration is self-consistent."""
+        self.localSetUp(mockFWHMs=[self.avgFwhm for i in range(self.config.nImagesMax)])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), self.config.nImagesMax)
+
+    def testNImagesLessThanNMax(self):
+        """Test case of input selectDataList shorter than nImagesMax."""
+        self.config.nImagesMax = 5
+        self.localSetUp(mockFWHMs=[self.avgFwhm for i in range(3)])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), 3)
+
+    def testNImagesGreaterThanNMax(self):
+        """Test case of input selectDataList greater than nImagesMax."""
+        self.localSetUp(mockFWHMs=[self.avgFwhm for i in range(self.config.nImagesMax+3)])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), self.config.nImagesMax)
+
+    def testFwhmSelection(self):
+
+        self.config.minPsfFwhm = 1.0
+        self.config.maxPsfFwhm = 1.5
+        self.config.nImagesMax = 10
+        self.localSetUp(mockFWHMs=[0.5, 1.0, 1.25, 1.5, 2.0])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList, selectDataList=self.selectList)
+        self.assertEqual(len(result.exposureInfoList), 3)
+
+    def testSelectDefaultMakeDatarRefListIsFalse(self):
+        self.localSetUp(mockFWHMs=[self.avgFwhm for i in range(self.config.nImagesMax)])
+        task = selectImages.BestSeeingWcsSelectImagesTask(config=self.config)
+        result = task.runDataRef(self.dataRef, self.coordList,
+                                 selectDataList=self.selectList, makeDataRefList=False)
+        self.assertIsNone(result.dataRefList)
+
```

Comment:
These test cases are marginally interesting, but what I'd really like to see is a slightly more realistic case where more than Nmax images meet the good seeing criterion but have different seeings. For example, if there are 10 images with acceptable FWHM values, but they aren't all the same, and Nmax = 3, I think the desired behavior would be for the selector to choose the three with the lowest FWHM values. This requires testing more than just the length of the resulting exposure list, but rather what FWHM values and/or dataIds made it into the list.